tools/ci_build/github/apple/test_ios_packages.py

```diff
@@ -0,0 +1,100 @@
+        raise FileNotFoundError('c_framework_dir {} is not a folder.'.format(c_framework_dir))
+
+    framework_path = os.path.join(c_framework_dir, 'onnxruntime.framework')
+    if not pathlib.Path(framework_path).exists():
+        raise FileNotFoundError('{} does not have onnxruntime.framework'.format(c_framework_dir))
+
+    # create a temp folder under repo dir
+    import tempfile
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # temp_dir = '/Users/gwang/temp/eee'
+
+        # create a zip file contains the framework
+        local_pods_dir = os.path.join(temp_dir, 'local_pods')
+        os.makedirs(local_pods_dir, exist_ok=True)
+        # shutil.make_archive require target file as full path without extension
+        zip_base_filename = os.path.join(local_pods_dir, 'OnnxRuntimeBase')
+        zip_file_path = zip_base_filename + '.zip'
+        shutil.make_archive(zip_base_filename, 'zip', root_dir=c_framework_dir, base_dir='onnxruntime.framework')
+
+        # copy the test project to the temp_dir
+        test_proj_path = os.path.join(REPO_DIR, 'onnxruntime', 'test', 'platform', 'ios', 'ios_package_test')
+        target_proj_path = os.path.join(temp_dir, 'ios_package_test')
+        if pathlib.Path(target_proj_path).exists():
+            shutil.rmtree(target_proj_path)
+        shutil.copytree(test_proj_path, target_proj_path)
+
+        # update the podspec to point to the local framework zip file
+        local_podspec_path = os.path.join(target_proj_path, 'OnnxRuntimeBase.podspec')
+        with open(local_podspec_path, 'r') as file:
+            file_data = file.read()
+
+        # replace the target strings
+        file_data = file_data.replace('${ORT_BASE_FRAMEWORK_FILE}', zip_file_path)
+        with open(os.path.join(REPO_DIR, 'VERSION_NUMBER')) as version_file:
+            file_data = file_data.replace('${ORT_VERSION}', version_file.readline().strip())
+
+        # overwrite the file
+        with open(local_podspec_path, 'w') as file:
+            file.write(file_data)
+
+        # install pods first
+        subprocess.run(['pod', 'install'], shell=False, check=True, cwd=target_proj_path)
+
+        # run the tests
+        subprocess.run(['xcrun', 'xcodebuild', 'test', '-quiet',
+                        '-workspace', './ios_package_test.xcworkspace',
+                        '-scheme', 'ios_package_test',
+                        '-destination', 'platform=iOS Simulator,OS=latest,name=iPhone SE (2nd generation)'],
+                       shell=False, check=True, cwd=target_proj_path)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        os.path.basename(__file__),
+        description='Test iOS framework using CocoaPods package.'
+    )
+
+    parser.add_argument('--fail_if_cocoapods_missing', action='store_true',
+                        help='This script will fail if CocoaPods is not installed, '
+                        'will not throw error unless fail_if_cocoapod_missing is set.')
```

Comment:
I'm a bit unclear about what this option does. if cocoapods is not installed, will this script always fail, or only fail if this option is set?

Reply:
We will set this option to True in our packaging pipeline (not exist yet), since the Azure CI machine will have CocoaPods installed, and we want to make sure these tests run

But if you are building and test on your own machine, the build-test should not fail if you don't have CocoaPods installed

Reply:
how about something like this:
"Whether to fail or skip the test if CocoaPods is not installed."

Reply:
Will update in my next PR